### DISCUSSION TITLE
How to format XLSX cells?

### DISCUSSION THREAD

**Discussion by ondohotola** [dathere/qsv#983] [Q&A] [answered]
I have to generate a really messed up XLSX file for monthly upload to the Revenue Service (Employee's Tax). 

It has 81 columns, a header row, and a row for each employee, but that is not the issue.

It also uses an extremely weird format for the cells (columns) containing amounts.

```excel
_([$NAD] * #,##0.00_);_([$NAD] * (#,##0.00);_([$NAD] * "-"??_);_(@_)
```

I would like to be able to format the cells containing amounts (ie columns) using the (**any**) format string which Excel shows when you look at the cell format, which in our case is the above.

Currently I am investigating a combination of perl Excel Reader/Writer modules.

I this something having been considered by someone before and perhaps someone already has a solution?

**Comment by jqnatividad** [comment, marked as answer]
Hi @ondohotola ,
First off, I moved your query to Discussions so other people who may have the same question can more easily find it in the future.

To your query, we do have the `to xlsx` subcommand, but that doesn't handle your current situation.

Though it uses the [`xlsxwriter`](https://docs.rs/xlsxwriter/latest/xlsxwriter/index.html) crate that does support specifying both formats and formulas, I think its out of scope for the `to` command.

What you may want to do is add either `luau` or `py` to your pipeline and implement the format there, before converting the CSV to xlsx.

Python's [f-string expressions](https://zetcode.com/python/fstring/) are particularly powerful and easy to use.

Luau's [string interpolation](https://luau-lang.org/2023/02/02/luau-string-interpolation.html) feature can also be leveraged for this issue.
